packages/sdk/src/sweep-and-flip/order.ts

```diff
@@ -0,0 +1,126 @@
+import { Provider } from "@ethersproject/abstract-provider";
+import { _TypedDataEncoder } from "@ethersproject/hash";
+import { Signer } from "ethers";
+
+import * as Addresses from "./addresses";
+import * as Types from "./types";
+import * as Common from "../common";
+import { bn, getCurrentTimestamp, lc, n, s, uniqBy } from "../utils";
+import { OrderType } from "./types";
+import { isValidOrderType, isValidSwap } from "./helpers";
+
+export class Order {
+  public chainId: number;
+  public params: Types.SwapOrderParams;
+
+  constructor(chainId: number, params: Types.SwapOrderParams) {
+    this.chainId = chainId;
+
+    try {
+      this.params = normalize(params)
+    } catch {
+      throw new Error("Invalid params");
+    }
+
+    if (!params.orderType) {
+      throw new Error("Empty order type");
+    }
```

Comment:
Probably not, because we will set the orderType in orderbook and/or router but I think we can leave there just to ensure that the orderType is not empty